VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -75,12 +71,20 @@ public void updateDrawState(TextPaint ds) {
                         Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
             }
 
-            text = trim(ResourceUtil.getFormattedString(context.getResources(),
-                    stringRes, new HashMap<String, CharSequence>() {{
-                        put("time", formattedTime);
-                        put("author", authorSpan);
-                        put("author_label", authorLabelFormatted.toString());
-                    }}));
+            if (authorLabel != null) {
+                text = trim(ResourceUtil.getFormattedString(context.getResources(),
+                        stringRes, new HashMap<String, CharSequence>() {{
+                            put("time", formattedTime);
+                            put("author", authorSpan);
+                            put("author_label", authorLabel);
+                        }}));
+            } else {
+                text = trim(ResourceUtil.getFormattedString(context.getResources(),
+                        stringResWithoutLabel, new HashMap<String, CharSequence>() {{
+                            put("time", formattedTime);
+                            put("author", authorSpan);
+                        }}));
```

Comment:
Since just the key/value `HashMap` is different in the two scenarios, it might make sense to initialize it into a separate variable, and just add the author label to the map conditional on it being non-null, and calling the method separately thereafter.